dockersetup.py

```diff
@@ -0,0 +1,904 @@
+            else:
+                newlines.append(format + '#- "EMAIL_PORT=<set this here if not the default>"\n')
+        elif '- "EMAIL_USER' in line:
+            format = line[0:line.find('- "EMAIL_USER')].replace("#", "")
+            if email_user:
+                newlines.append(format + '- "EMAIL_USER=' + email_user + '"\n')
+            else:
+                newlines.append(format + '#- "EMAIL_USER=<set this here if needed>"\n')
+        elif '- "EMAIL_PASSWORD' in line:
+            format = line[0:line.find('- "EMAIL_PASSWORD')].replace("#", "")
+            if email_password:
+                newlines.append(format + '- "EMAIL_PASSWORD=' + email_password + '"\n')
+            else:
+                newlines.append(format + '#- "EMAIL_PASSWORD=<set this here if needed>"\n')
+        elif '- "EMAIL_USE_SSL' in line:
+            format = line[0:line.find('- "EMAIL_USE_SSL')].replace("#", "")
+            if email_ssl:
+                newlines.append(format + '- "EMAIL_USE_SSL=' + email_ssl + '"\n')
+            else:
+                newlines.append(format + '#- "EMAIL_USE_SSL=<set this here if needed>"\n')
+        elif '- "EMAIL_USE_TLS' in line:
+            format = line[0:line.find('- "EMAIL_USE_TLS')].replace("#", "")
+            if email_tls:
+                newlines.append(format + '- "EMAIL_USE_TLS=' + email_tls + '"\n')
+            else:
+                newlines.append(format + '#- "EMAIL_USE_TLS=<set this here if needed>"\n')
+        elif "ports:" in line:
+            if in_layersweb:
+                in_layersweb_ports = True
+            newlines.append(line + "\n")
+        elif "./docker/certs:/" in line:
+            newlines.append(adjust_cert_mount_line(line) + '\n')
+        else:
+            newlines.append(line + "\n")
+    writefile("docker-compose.yml", ''.join(newlines))
+
+
+def read_nginx_ssl_conf(certdir):
+    hostname = None
+    https_port = None
+    certdir = None
+    certfile = None
+    keyfile = None
+    with open('docker/nginx-ssl-edited.conf', 'r') as f:
+        for line in f:
+            if 'ssl_certificate ' in line:
+                certdir, certfile = os.path.split(line.split('ssl_certificate', 1)[1].strip().rstrip(';'))
+            elif 'ssl_certificate_key ' in line:
+                keyfile = os.path.basename(line.split('ssl_certificate_key', 1)[1].strip().rstrip(';'))
+            elif 'server_name ' in line:
+                sname = line.split('server_name', 1)[1].strip().rstrip(';')
+                if sname != '_':
+                    hostname = sname
+            elif 'return 301 https://' in line:
+                res = re.search(':([0-9]+)', line)
+                if res:
+                    https_port = res.groups()[0]
+    ret = (hostname, https_port, certdir, certfile, keyfile)
+    if None in ret:
+        sys.stderr.write('Failed to read SSL configuration from nginx-ssl-edited.conf')
+        sys.exit(1)
+    return ret
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears you are unpacking more than three variables while using a function that returns more than one variable. Using a large number of return values is prone to errors. We recommend that you return a small class or `namedtuple` instance instead.

[Learn more](https://docs.python.org/3/library/typing.html#typing.NamedTuple)